### DISCUSSION TITLE
Dummy instrument with fast traces

### DISCUSSION THREAD

**Discussion by EdwardLaird1** [microsoft/Qcodes#6072] [Q&A]
Does there exist a dummy instrument that simulates measurement of an entire trace (as one would do from an oscilloscope or spectrum analyser, for example)?
This would be useful for testing measurement scripts, just as DummyInstrumentWithMeasurement is useful for testing measurements of individual data points.

**Comment by jenshnielsen** [comment]
The DummyChannel bound to DummyChannelInstrument has a 1d and a 2d ParameterWithSetpoints dummy_parameter_with_setpoints, and dummy_parameter_with_setpoints_2d see https://github.com/microsoft/Qcodes/blob/main/src/qcodes/instrument_drivers/mock_instruments/__init__.py#L301
